README.md

```diff
@@ -1017,6 +1027,41 @@ directives (see [Modules](#modules)).
   end
   ```
 
+### Callbacks
+
+Callbacks are used to define [Behaviours].
+
+Callbacks should be defined at the top of your module, right before any
+function (see [Modules](#modules)).
+
+* <a name="callbacks"></a>
+  Each `@callback` or `@macrocallback` should be grouped with its corresponding
+  `@doc`. Separate each pair with a blank line.
+
+  ```elixir
+  @doc "Describe what the callback function should do"
+  @callback operation() :: :ok | {:error, term}
+
+  @doc "Describe what the macro is expected to do"
+  @macrocallback macro_name() :: Macro.t
+  ```
+
+* <a name="callback-types"></a>
+  The types on callbacks should follow the rules for [Typespecs](#typespecs).
+
+* <a name="optional-callbacks"></a>
+  Optional callbacks should be listed last right before `@optional_callbacks`:
```

Comment:
I'm starting to wonder if people need this much detail. The order is implied in the Modules section. The `@doc` and typespec rules are not special for callbacks.

Would you be willing to break this into two commits? I can cherry pick the module ordering one, and we can save the Callbacks section for another time.